### ISSUE TITLE
Use file-like file type recognition instead of file extension

### ISSUE THREAD

**Issue by vertbyqb** [lbryio/lbry-sdk#3481]
When publishing to LBRY with `./lbrynet publish`, it detects the file type by the file extension, which can lead to errors, like if a file has the wrong extension or there is no extension. Instead it should use a Unix `file` (https://en.wikipedia.org/wiki/File_(command)) like approach.

# Reproduction

1. Copy `lbrynet` binary to an empty directory.
2. `cp lbrynet a.md`
3. `./lbrynet publish --name=upload-for-bug-report --bid=0.001 --file_name=a.md`

# Output
```
...
"value": {
        "source": {
          "media_type": "text/markdown",
          "name": "a.md"
        },
...
```

# Expected output
```
...
"value": {
        "source": {
          "media_type": "application/octet-stream",
          "name": "a.md"
        },
...
```

# Version

```
$ ./lbrynet --version
lbrynet 0.105.0
```

**Comment by belikor** [comment]
You are right, the current code essentially guesses the file type when the claim is created.
https://github.com/lbryio/lbry-sdk/blob/d19e07d66101c4e3ac9c0cb9030a6eb098cd51a1/lbry/schema/claim.py#L236-L241

The media types are defined here:
https://github.com/lbryio/lbry-sdk/blob/master/lbry/schema/mime_types.py

**Comment by eug3nix** [comment]
Hi. I would like to take this one as a "good first issue". 
There is a number of libraries which can detect filetype by its content, mainly:
https://github.com/h2non/filetype.py which is a pure-python implmentation
and 
https://github.com/ahupp/python-magic which is a binding to the libmagic, the same library which the mentioned *file* command uses.
Q: I would like to know what is the general approach in lbry regarding external dependencies (expecially native bindings vs pure-python implementations in this case).

Cheers!

**Comment by belikor** [comment]
> Q: I would like to know what is the general approach in lbry regarding external dependencies (expecially native bindings vs pure-python implementations in this case).

I think if you use a pure Python library it's better. Just write your solution and submit it, then it will be reviewed.
